assets/app/view/train_roster.rb

```diff
@@ -27,32 +27,115 @@ def render_body
         ])
       end
 
+      children << phases
+
       h(:div, {}, children)
     end
 
+    def cert_limit
+      cert_props = {
+        style: {
+          'margin-bottom': '1rem'
+        }
+      }
+      h(:div, cert_props, "Certificate limit: #{@game.cert_limit}")
+    end
+
+    def phases
+      td_props = {
+        style: {
+          padding: '0 1rem'
+        }
+      }
+
+      current_phase = @game.phase.current
+      rows = @game.phase.phases.map do |phase|
+        phase_color = Array(phase[:tiles]).last
+        phase_props = {
+          style: {
+            padding: '0 1rem'
+          }
+        }
+        if View::Part::MultiRevenue::COLOR.include? phase_color
+          m = View::Part::MultiRevenue::COLOR[phase_color].match(/#(..)(..)(..)/)
+          phase_props[:style]['background-color'] = "rgba(#{m[1].hex},#{m[2].hex},#{m[3].hex},0.4)"
+        end
+
+        buy_text = if phase[:buy_companies]
+                     'Can Buy'
+                   elsif phase[:events]&.include?(:close_companies)
+                     'Close'
+                   else
+                     ''
+                   end
+
+        h(:tr, [
+          h(:td, td_props, phase[:name] + (current_phase == phase ? ' (Current) ' : '')),
+          h(:td, td_props, phase[:operating_rounds]),
+          h(:td, td_props, phase[:train_limit]),
+          h(:td, phase_props, phase_color.capitalize),
+          h(:td, td_props, buy_text)
+        ])
+      end
+
+      props = {
+        style: { 'margin-top': '1rem' }
+      }
+      h(:div, props, [
+        h(:div, 'Game Phases'),
+        h(:table, [
+          h(:tr, [
+            h(:th, td_props, 'Phase'),
+            h(:th, td_props.merge({attrs: {title:'Number of Operating Rounds'}}), '# OR'),
+            h(:th, td_props, 'Train Limit'),
+            h(:th, td_props, 'Tiles'),
+            h(:th, td_props, 'Companies')
+          ]),
+          *rows
+        ])
+      ])
+    end
+
     def upcoming_trains
       td_props = {
         style: {
           padding: '0 1rem'
         }
       }
 
+      rust_schedule = {}
+      @depot.trains.group_by(&:name).each do |name, trains|
```

Comment:
i feel like this could be written better

Reply:
Yeah, i'd agree but i'm not sure the best rubyism. If it wasn't creating a list of lists .to_h would make sense, but doesn't in this.